src/ILCompiler.WebAssembly/src/CodeGen/ILToWebAssemblyImporter.cs

```diff
@@ -2701,6 +2714,47 @@ private void ImportThrow()
             EmitTrapCall();
         }
 
+        private void ThrowIfNull(LLVMValueRef entry)
+        {
+            if (NullRefFunction.Pointer == IntPtr.Zero)
+            {
+                NullRefFunction = LLVM.AddFunction(Module, "corert.throwifnull", LLVM.FunctionType(LLVM.VoidType(), new LLVMTypeRef[] { LLVM.PointerType(LLVMTypeRef.Int8Type(), 0), LLVM.PointerType(LLVMTypeRef.Int8Type(), 0) }, false));
```

Comment:
Consider adding the ```inlinehint``` attribute to this function (you could use ```alwaysinline```, but I'm not sure we always know better than the compiler)

Reply:
I cant seem to find the value for inlinehint, I'm assuming I would set it using LLVM.AddFunctionAttr2 but there isn't an enum in LLVMSharp. Do you know where I should be looking?

Reply:
I did some digging and it appears that the answer is here https://github.com/llvm-mirror/llvm/blob/993ef0ca960f8ffd107c33bfbf1fd603bcf5c66c/include/llvm/IR/Attributes.td but this file ends up generating the real enum so I'm not certain I should be trusting the value I come up with. I think It would need to be explicitly supported by llvmsharp in order to not break in terrible ways on upgrade of LLVM.